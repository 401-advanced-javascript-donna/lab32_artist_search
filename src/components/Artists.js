import React from 'react';
import PropTypes from 'prop-types';
import Artist from './Artist';

const Artists = ({ artistArray }) => {
  
  const artists = artistArray.map(artist => {
    return <Artist
      key={artist.id}
      name={artist.name}
      type={artist.type}
      gender={artist.gender}
      tags={artist.tags}
    />;
  });

  return (
    <ul>
      {artists}
    </ul>
  );
};

Artists.propTypes = {
  artistArray: PropTypes.array.isRequired
};

export default Artists;
