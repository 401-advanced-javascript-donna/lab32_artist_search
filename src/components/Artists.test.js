import React from 'react';
import { shallow } from 'enzyme';
import Artists from './Artists';

describe('Artists component', () => {
  it('renders Artists', () => {
    const wrapper = shallow(<Artists artistArray={[{ id: 'Tom Petty' }, { id: 'Frank Zappa' }]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
