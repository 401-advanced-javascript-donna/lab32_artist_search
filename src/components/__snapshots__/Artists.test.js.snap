// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists component renders Artists 1`] = `
<ul
  className="Artists"
>
  <Artist
    key="Tom Petty"
  />
  <Artist
    key="Frank Zappa"
  />
</ul>
`;
